src/sentry_value.c

```diff
@@ -806,6 +807,38 @@ sentry_value_is_null(sentry_value_t value)
     return value._bits == CONST_NULL;
 }
 
+int
+sentry__value_merge_objects(sentry_value_t dst, sentry_value_t src)
+{
+    if (sentry_value_get_type(dst) != SENTRY_VALUE_TYPE_OBJECT
+        || sentry_value_get_type(src) != SENTRY_VALUE_TYPE_OBJECT
+        || sentry_value_is_frozen(dst)) {
+        return 1;
```

Comment:
oh, I think the problem with the test assertions is that the event (aka `dst`) does not have any tags/extra set at all, so this short circuits.

Reply:
hum, maybe it should be a graceful no-op if the src is null?

Reply:
I think that's justifiable. What should happen if `dst` is null? Replace it wholesale with `src`?

Reply:
You could also keep this function's semantics "pure" and add a wrapper that handles the cases of `src` or `dst` being null.

Reply:
yeah, i think `src` being null is fine in this function.  but for `dst` being null there's not much you can do as the caller likely has other shenanigans to deal with